tracer/test/snapshots/GraphQL7Tests.SubmitsTraces.verified.txt

```diff
@@ -0,0 +1,451 @@
+      http.method: POST,
+      http.request.headers.host: localhost:00000,
+      http.status_code: 405,
+      http.url: http://localhost:00000/graphql,
+      http.useragent: testhelper,
+      language: dotnet,
+      runtime-id: Guid_1,
+      span.kind: server,
+      version: 1.0.0,
+      _dd.p.dm: -0
+    },
+    Metrics: {
+      process_id: 0,
+      _dd.agent_psr: 1.0,
+      _dd.top_level: 1.0,
+      _dd.tracer_kr: 1.0,
+      _sampling_priority_v1: 1.0
+    }
+  },
+  {
+    TraceId: Id_7,
+    SpanId: Id_17,
+    Name: graphql.validate,
+    Resource: graphql.validate,
+    Service: Samples.GraphQL7-graphql,
+    Type: graphql,
+    ParentId: Id_8,
+    Tags: {
+      component: GraphQL,
+      env: integration_tests,
+      graphql.source: query HeroQuery{hero {name appearsIn}},
+      language: dotnet,
+      runtime-id: Guid_1,
+      span.kind: server
+    },
+    Metrics: {
+      _dd.top_level: 1.0
+    }
+  },
+  {
+    TraceId: Id_7,
+    SpanId: Id_18,
+    Name: graphql.execute,
+    Resource: Query HeroQuery,
+    Service: Samples.GraphQL7-graphql,
+    Type: graphql,
+    ParentId: Id_8,
+    Tags: {
+      component: GraphQL,
+      env: integration_tests,
+      graphql.operation.name: HeroQuery,
+      graphql.operation.type: Query,
+      graphql.source: query HeroQuery{hero {name appearsIn}},
+      language: dotnet,
+      runtime-id: Guid_1,
+      span.kind: server
+    },
+    Metrics: {
+      _dd.top_level: 1.0
+    }
+  },
+  {
+    TraceId: Id_9,
+    SpanId: Id_19,
+    Name: graphql.validate,
+    Resource: graphql.validate,
+    Service: Samples.GraphQL7-graphql,
+    Type: graphql,
+    ParentId: Id_10,
+    Tags: {
+      component: GraphQL,
+      env: integration_tests,
+      graphql.source: mutation AddBobaFett($human:HumanInput!){createHuman(human: $human){id name}},
+      language: dotnet,
+      runtime-id: Guid_1,
+      span.kind: server
+    },
+    Metrics: {
+      _dd.top_level: 1.0
+    }
+  },
+  {
+    TraceId: Id_9,
+    SpanId: Id_20,
+    Name: graphql.execute,
+    Resource: Mutation AddBobaFett,
+    Service: Samples.GraphQL7-graphql,
+    Type: graphql,
+    ParentId: Id_10,
+    Tags: {
+      component: GraphQL,
+      env: integration_tests,
+      graphql.operation.name: AddBobaFett,
+      graphql.operation.type: Mutation,
+      graphql.source: mutation AddBobaFett($human:HumanInput!){createHuman(human: $human){id name}},
+      language: dotnet,
+      runtime-id: Guid_1,
+      span.kind: server
+    },
+    Metrics: {
+      _dd.top_level: 1.0
+    }
+  },
+  {
+    TraceId: Id_11,
+    SpanId: Id_21,
+    Name: graphql.validate,
+    Resource: graphql.validate,
+    Service: Samples.GraphQL7-graphql,
+    Type: graphql,
+    ParentId: Id_12,
+    Error: 1,
+    Tags: {
+      component: GraphQL,
+      env: integration_tests,
+      error.msg: 1 error(s),
+      error.stack:
+errors: [
+{
+"message": "Subscription operations are not supported for POST requests.",
```

Comment:
is this one normal 🤔

Reply:
That's a new change from the latest version (7). You can't do a POST request on subscription.
That's the error returned by GraphQL using the same request as the GraphQL(4&5) sample.

Reply:
Do GraphQL docs give a reason for this change? Do they suggest a migration path? If so perhaps we should add a test for that.

Reply:
indeed couldnt' find anything in the breaking changes https://graphql-dotnet.github.io/docs/migrations/migration7/  or anywhere online

Reply:
Couldn't find anything either but subscriptions is done with websocket (I don't know if there are other methods).
I can see here that the check was added since version 7.0.0 (from tags): https://github.com/graphql-dotnet/server/blob/9cc63468f5fb4b189d4144c2a1c2e04401b3457c/src/Transports.AspNetCore/HttpPostValidationRule.cs

Reply:
Yes, we need to rewrite our integration tests to use websockets. This has been the blocker on upgrading the 7.0 support!